iptables/busybox are mounted instead of copied

tobsan opened the pull request:
Also adds proc and sys as lxc.mount.auto entries.

Plus some general cleanup in the LXC template

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

frznlogic commented on libsoftwarecontainer/lxc-softwarecontainer.in:
```diff
@@ -56,10 +54,10 @@ create_rootfs()
         $rootfs/tmp \
         $rootfs/gateways \
-        $rootfs/${CMAKE_INSTALL_PREFIX} \
-        ${ADDITIONAL_FOLDER_MOUNTS}"
```
I assume we have no use for this parameter?

tobsan replied:
We don't use it anywhere in the rest of the code base.

frznlogic replied:
Ok, I was just considering if it was something that the integrator/user would like to have at some point.

DunderRoffe reviewed (changes requested):
At a quick glance the content seems nice. However, this should really be broken up in several commits.

frznlogic commented on libsoftwarecontainer/lxc-softwarecontainer.in:
```diff
@@ -69,38 +67,53 @@ create_rootfs()
+    touch $rootfs/bin/init.lxc
+    chmod 755 $rootfs/bin/init.lxc
+
+    # If we have network enabled, we need iptables
```
Do we really _always_ need it? I'm not going to nitpick about it i guess.

tobsan replied:
Yes, if we have network gateway enabled, we will **always** need it.

frznlogic commented on libsoftwarecontainer/lxc-softwarecontainer.in:
```diff
@@ -109,50 +122,37 @@ copy_configuration()
+options=$(getopt -o p:n: -l path:,name:,rootfs: -- "$@")
 if [ $? -ne 0 ]; then
-    usage $(basename $0)
+    echo "Usage: $(basename $0) -p|--path=<path> --rootfs=<path>"
```
Why did we remove the usage function? Imho, it could have been expanded with more information and probably looked better than this.

tobsan replied:
This template is called from within LXC, so imho this is enough printouts for whoever encounters a problem.

frznlogic reviewed (approved):
I approve with some comments and questions.

Commit:
Set "lxc.mount.auto = proc sys" in the LXC config

This auto-mounts proc and sys with reasonable defaults inside the
container. Also a small cleanup of things that are all below /usr

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

Commit:
Bind-mount busybox and iptables instead of copy

This saves some space in the container and we won't have
to bind-mount all of /sbin into the container, which is nice
to avoid if we can.

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

Commit:
Some LXC template cleanup

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

DunderRoffe reviewed (approved):
Assuming you fix the comments from @frznlogic this is ok
